lib/index.js

```diff
@@ -3,65 +3,46 @@
 const spawn = require('cross-spawn').spawn
 const path = require('path')
 const fs = require('fs')
+const ParseArgs = require('./parse-args')
 
 function EnvCmd (args) {
   // Parse the args from the command line
   const parsedArgs = ParseArgs(args)
-
-  // Attempt to open the provided file
-  let file
-  try {
-    file = fs.readFileSync(parsedArgs.envFilePath, { encoding: 'utf8' })
-  } catch (e) {
-    throw new Error(`Error! Could not find or read file at ${parsedArgs.envFilePath}`)
-  }
-
-  // Parse the env file string
-  const env = path.extname(parsedArgs.envFilePath).toLowerCase() === '.json'
-    ? Object.assign({}, process.env, require(parsedArgs.envFilePath))
-    : ParseEnvString(file)
-
-  // Execute the command with the given environment variables
-  if (parsedArgs.command) {
-    const proc = spawn(parsedArgs.command, parsedArgs.commandArgs, {
-      stdio: 'inherit',
-      env
-    })
-    process.on('SIGTERM', () => proc.kill('SIGTERM'))
-    proc.on('exit', process.exit)
-    return proc
-  }
-}
-
-function ParseArgs (args) {
-  if (args.length < 2) {
-    throw new Error('Error! Too few arguments passed to env-cmd.')
-  }
-
-  let envFilePath
-  let command
-  let commandArgs = args.slice()
-  while (commandArgs.length) {
-    const arg = commandArgs.shift()
-
-    // assume the first arg is the env file
-    if (!envFilePath) {
-      envFilePath = path.resolve(process.cwd(), arg)
-    } else {
-      command = arg
-      break
+  const files = parsedArgs.files
+  const command = parsedArgs.command
+  const commandArgs = parsedArgs.commandArgs
+
+  // For all files..
+  let env = files.reduce((env, file) => {
+    try {
+      // Attempt to open the provided file
+      const envFileText = fs.readFileSync(file, 'utf8')
```

Comment:
I am thinking maybe this chunk (inside the `try...catch`) should be separated out into its own function (maybe even a separate file?) so that its clear where we read in the different file types. By separating it out, it puts us in a position to more easily support even more file types in the future if so desired. Not only that, but we can more easily test this separate function which is pretty much the core business logic in this app 😉